diff --git a/cross-ride2 b/cross-ride2
new file mode 160000
index 0000000..8f597a6
--- /dev/null
+++ b/cross-ride2
@@ -0,0 +1 @@
+Subproject commit 8f597a6e0572e29c68d998d867f851c9dc3a621c
diff --git a/cross-ride2/.classpath b/cross-ride2/.classpath
deleted file mode 100644
index 1e0aee2..0000000
--- a/cross-ride2/.classpath
+++ /dev/null
@@ -1,80 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<classpath>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
-	<classpathentry kind="src" path="src"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-aop-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-aop-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-aop-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-aspects-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-aspects-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-aspects-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-beans-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-beans-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-beans-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-context-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-context-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-context-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-context-indexer-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-context-indexer-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-context-indexer-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-context-support-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-context-support-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-context-support-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-core-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-core-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-core-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-expression-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-expression-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-expression-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-instrument-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-instrument-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-instrument-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-jcl-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-jcl-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-jcl-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-jdbc-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-jdbc-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-jdbc-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-jms-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-jms-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-jms-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-messaging-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-messaging-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-messaging-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-orm-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-orm-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-orm-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-oxm-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-oxm-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-oxm-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-test-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-test-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-test-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-tx-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-tx-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-tx-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-web-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-web-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-web-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-webflux-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-webflux-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-webflux-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-webmvc-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-webmvc-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-webmvc-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-websocket-5.1.0.RELEASE-javadoc.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-websocket-5.1.0.RELEASE-sources.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-framework-5.1.0.RELEASE/libs/spring-websocket-5.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-boot-autoconfigure-2.0.2.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-boot-1.2.3.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/log4j-1.2.17.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/slf4j-api-2.0.99.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/slf4j-simple-1.6.99.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-data-commons-1.6.1.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-data-commons-core-1.1.0.RELEASE.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/javax.persistence.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/validation-api-2.0.0.Final.jar"/>
-	<classpathentry kind="lib" path="/Users/thefavourofpraise/Documents/javalib/spring-data-rest-core-2.0.0.RELEASE.jar"/>
-	<classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/4"/>
-	<classpathentry kind="output" path="bin"/>
-</classpath>
diff --git a/cross-ride2/.gitignore b/cross-ride2/.gitignore
deleted file mode 100644
index ae3c172..0000000
--- a/cross-ride2/.gitignore
+++ /dev/null
@@ -1 +0,0 @@
-/bin/
diff --git a/cross-ride2/.project b/cross-ride2/.project
deleted file mode 100644
index ac0591a..0000000
--- a/cross-ride2/.project
+++ /dev/null
@@ -1,17 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<projectDescription>
-	<name>cross-ride2</name>
-	<comment></comment>
-	<projects>
-	</projects>
-	<buildSpec>
-		<buildCommand>
-			<name>org.eclipse.jdt.core.javabuilder</name>
-			<arguments>
-			</arguments>
-		</buildCommand>
-	</buildSpec>
-	<natures>
-		<nature>org.eclipse.jdt.core.javanature</nature>
-	</natures>
-</projectDescription>
diff --git a/cross-ride2/src/com/crossover/junit/CrossRideTest.java b/cross-ride2/src/com/crossover/junit/CrossRideTest.java
deleted file mode 100644
index f61a9df..0000000
--- a/cross-ride2/src/com/crossover/junit/CrossRideTest.java
+++ /dev/null
@@ -1,38 +0,0 @@
-package com.crossover.junit;
-
-import static org.junit.Assert.*;
-
-import org.junit.After;
-import org.junit.AfterClass;
-import org.junit.Before;
-import org.junit.BeforeClass;
-import org.junit.Test;
-
-import com.crossover.techtrial.service.PersonService;
-import com.crossover.techtrial.service.RideService;
-
-public class CrossRideTest {
-
-	@BeforeClass
-	public static void setUpBeforeClass() throws Exception {
-	}
-
-	@AfterClass
-	public static void tearDownAfterClass() throws Exception {
-	}
-
-	@Before
-	public void setUp() throws Exception {
-	}
-
-	@After
-	public void tearDown() throws Exception {
-	}
-
-	@Test
-	public void test() {
-		//fail("Not yet implemented");
-		assertEquals(null, "1234455");
-	}
-
-}
diff --git a/cross-ride2/src/com/crossover/techtrial/CrossRideApplication.java b/cross-ride2/src/com/crossover/techtrial/CrossRideApplication.java
deleted file mode 100644
index e619ea0..0000000
--- a/cross-ride2/src/com/crossover/techtrial/CrossRideApplication.java
+++ /dev/null
@@ -1,15 +0,0 @@
-package com.crossover.techtrial;
-
-import org.springframework.boot.SpringApplication;
-import org.springframework.boot.autoconfigure.SpringBootApplication;
-
-/**
- * @author crossover
- *
- */
-@SpringBootApplication
-public class CrossRideApplication {
-  public static void main(String[] args) {
-    SpringApplication.run(CrossRideApplication.class, args);
-  }
-}
diff --git a/cross-ride2/src/com/crossover/techtrial/controller/PersonController.java b/cross-ride2/src/com/crossover/techtrial/controller/PersonController.java
deleted file mode 100644
index 9a6bec8..0000000
--- a/cross-ride2/src/com/crossover/techtrial/controller/PersonController.java
+++ /dev/null
@@ -1,95 +0,0 @@
-/**
- * 
- */
-package com.crossover.techtrial.controller;
-
-import java.text.ParseException;
-import java.text.SimpleDateFormat;
-import java.util.Date;
-import java.util.List;
-
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
-
-import com.crossover.techtrial.model.Person;
-import com.crossover.techtrial.model.Ride;
-import com.crossover.techtrial.service.PersonService;
-import com.crossover.techtrial.service.RideService;
-
-/**
- * 
- * @author crossover
- */
-
-@RestController
-public class PersonController {
-  
-  @Autowired
-  PersonService personService;
-  RideService rideService;
-  
-  @PostMapping(path = "/api/person")
-  public ResponseEntity<Person> register(@RequestBody Person p) {
-	  
-    return ResponseEntity.ok(personService.save(p));
-  }
-  @PostMapping(path = "/api/ride")
-  public ResponseEntity<Person> submitRide(@RequestBody Ride r) {
-	  //check if the Rider and Person are valid entity 
-	  Person person = personService.findById(r.getDriver().getId());
-	    if (person == null) {
-	    	return ResponseEntity.notFound().build();
-	    }
-	    person = personService.findById(r.getRider().getId());
-	    if (person == null) {
-	    	return ResponseEntity.notFound().build();
-	    }
-	    
-	    try {
-			//r.getEndTime();
-			SimpleDateFormat dateFormatter = new SimpleDateFormat("dd/mm/yyyy");
-			//parse the date string to convert it into java.util.date object
-			Date startDate = dateFormatter.parse(r.getStartTime());
-			Date endDate = dateFormatter.parse(r.getEndTime());
-			
-			if(startDate.after(endDate) ||  startDate.equals(endDate)){
-				return ResponseEntity.notFound().build();
-			}
-			//
-			
-		} catch (Exception e) {
-			// TODO Auto-generated catch block
-			e.printStackTrace();
-		} 
-	    
-	    //check if driver already has at least 1 ride running
-	    //rideService.
-	    rideService.save(r);
-	    
-   
-    return null;
-  }
-  @GetMapping(path = "/api/top5driver")
-  public ResponseEntity<List<Person>> getTop5Persons() {
-    return ResponseEntity.ok(personService.getTop5Person());
-  }
-  @GetMapping(path = "/api/person")
-  public ResponseEntity<List<Person>> getAllPersons() {
-    return ResponseEntity.ok(personService.getAll());
-  }
-  
-  @GetMapping(path = "/api/person/{perso-id}")
-  public ResponseEntity<Person> getPersonById(@PathVariable(name="person-id", required=true)Long personId) {
-    Person person = personService.findById(personId);
-    if (person != null) {
-      return ResponseEntity.ok(person);
-    }
-    return ResponseEntity.notFound().build();
-  }
-  
-}
diff --git a/cross-ride2/src/com/crossover/techtrial/dto/TopDriverDTO.java b/cross-ride2/src/com/crossover/techtrial/dto/TopDriverDTO.java
deleted file mode 100644
index 9a807ea..0000000
--- a/cross-ride2/src/com/crossover/techtrial/dto/TopDriverDTO.java
+++ /dev/null
@@ -1,89 +0,0 @@
-/**
- * 
- */
-package com.crossover.techtrial.dto;
-
-/**
- * @author crossover
- *
- */
-public class TopDriverDTO {
-  
-  /**
-   * Constructor for TopDriverDTO
-   * @param name
-   * @param email
-   * @param totalRideDurationInSeconds
-   * @param maxRideDurationInSecods
-   * @param averageDistance
-   */
-  public TopDriverDTO(String name, 
-      String email, 
-      Long totalRideDurationInSeconds,
-      Long maxRideDurationInSecods,
-      Double averageDistance) {
-    this.setName(name);
-    this.setEmail(email);
-    this.setAverageDistance(averageDistance);
-    this.setMaxRideDurationInSecods(maxRideDurationInSecods);
-    this.setTotalRideDurationInSeconds(totalRideDurationInSeconds);
-    
-  }
-  
-  public TopDriverDTO() {
-    
-  }
-  
-  private String name;
-  
-  private String email;
-  
-  private Long totalRideDurationInSeconds;
-  
-  private Long maxRideDurationInSecods;
-  
-  private Double averageDistance;
-
-  public String getName() {
-    return name;
-  }
-
-  public void setName(String name) {
-    this.name = name;
-  }
-
-  public String getEmail() {
-    return email;
-  }
-
-  public void setEmail(String email) {
-    this.email = email;
-  }
-
-  public Long getTotalRideDurationInSeconds() {
-    return totalRideDurationInSeconds;
-  }
-
-  public void setTotalRideDurationInSeconds(Long totalRideDurationInSeconds) {
-    this.totalRideDurationInSeconds = totalRideDurationInSeconds;
-  }
-
-  public Long getMaxRideDurationInSecods() {
-    return maxRideDurationInSecods;
-  }
-
-  public void setMaxRideDurationInSecods(Long maxRideDurationInSecods) {
-    this.maxRideDurationInSecods = maxRideDurationInSecods;
-  }
-
-  public Double getAverageDistance() {
-    return averageDistance;
-  }
-
-  public void setAverageDistance(Double averageDistance) {
-    this.averageDistance = averageDistance;
-  }
-  
-  
-    
-}
diff --git a/cross-ride2/src/com/crossover/techtrial/exceptions/GlobalExceptionHandler.java b/cross-ride2/src/com/crossover/techtrial/exceptions/GlobalExceptionHandler.java
deleted file mode 100644
index e39127a..0000000
--- a/cross-ride2/src/com/crossover/techtrial/exceptions/GlobalExceptionHandler.java
+++ /dev/null
@@ -1,29 +0,0 @@
-package com.crossover.techtrial.exceptions;
-
-import java.util.AbstractMap;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
-import org.springframework.stereotype.Component;
-import org.springframework.web.bind.annotation.ControllerAdvice;
-import org.springframework.web.bind.annotation.ExceptionHandler;
-
-@ControllerAdvice
-@Component
-public class GlobalExceptionHandler {
-
-  private static final Logger LOG = LoggerFactory.getLogger(GlobalExceptionHandler.class);
-
-  /**
-   * Global Exception handler for all exceptions.
-   */
-  @ExceptionHandler
-  public ResponseEntity<AbstractMap.SimpleEntry<String, String>> handle(Exception exception) {
-    // general exception
-    LOG.error("Exception: Unable to process this request. ", exception);
-    AbstractMap.SimpleEntry<String, String> response =
-        new AbstractMap.SimpleEntry<>("message", "Unable to process this request.");
-    return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(response);
-  }
-}
diff --git a/cross-ride2/src/com/crossover/techtrial/model/Person.java b/cross-ride2/src/com/crossover/techtrial/model/Person.java
deleted file mode 100644
index 04d44f3..0000000
--- a/cross-ride2/src/com/crossover/techtrial/model/Person.java
+++ /dev/null
@@ -1,127 +0,0 @@
-/**
- * 
- */
-package com.crossover.techtrial.model;
-
-import java.io.Serializable;
-import javax.persistence.Column;
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.GenerationType;
-import javax.persistence.Id;
-import javax.persistence.Table;
-import javax.validation.constraints.Email;
-import javax.validation.constraints.NotNull;
-
-
-/**
- * @author crossover
- *
- */
-@Entity
-@Table(name = "person")
-public class Person implements Serializable{
-
-  private static final long serialVersionUID = 7401548380514451401L;
-  
-  public Person() {}
- 
-  @Id
-  @GeneratedValue(strategy = GenerationType.IDENTITY)
-  Long id;
-
-  @Column(name = "name")
-  String name;
-
-  @NotNull
-  @Email
-  @Column(name = "email")
-  String email;
-
-  @Column(name = "registration_number")
-  String registrationNumber;
-
-  public Long getId() {
-    return id;
-  }
-
-  public void setId(Long id) {
-    this.id = id;
-  }
-
-  public String getName() {
-    return name;
-  }
-
-  public void setName(String name) {
-    this.name = name;
-  }
-
-  public String getEmail() {
-    return email;
-  }
-
-  public void setEmail(String email) {
-    this.email = email;
-  }
-
-  public String getRegistrationNumber() {
-    return registrationNumber;
-  }
-
-  public void setRegistrationNumber(String registrationNumber) {
-    this.registrationNumber = registrationNumber;
-  }
-
-  @Override
-  public int hashCode() {
-    final int prime = 31;
-    int result = 1;
-    result = prime * result + ((email == null) ? 0 : email.hashCode());
-    result = prime * result + ((id == null) ? 0 : id.hashCode());
-    result = prime * result + ((name == null) ? 0 : name.hashCode());
-    result = prime * result + ((registrationNumber == null) ? 0 : registrationNumber.hashCode());
-    return result;
-  }
-
-  @Override
-  public boolean equals(Object obj) {
-    if (this == obj)
-      return true;
-    if (obj == null)
-      return false;
-    if (getClass() != obj.getClass())
-      return false;
-    Person other = (Person) obj;
-    if (email == null) {
-      if (other.email != null)
-        return false;
-    } else if (!email.equals(other.email))
-      return false;
-    if (id == null) {
-      if (other.id != null)
-        return false;
-    } else if (!id.equals(other.id))
-      return false;
-    if (name == null) {
-      if (other.name != null)
-        return false;
-    } else if (!name.equals(other.name))
-      return false;
-    if (registrationNumber == null) {
-      if (other.registrationNumber != null)
-        return false;
-    } else if (!registrationNumber.equals(other.registrationNumber))
-      return false;
-    return true;
-  }
-
-  @Override
-  public String toString() {
-    return "Person [id=" + id + ", name=" + name + ", email=" + email + ", registrationNumber=" + registrationNumber + "]";
-  }
-  
-  
-
-
-}
diff --git a/cross-ride2/src/com/crossover/techtrial/model/Ride.java b/cross-ride2/src/com/crossover/techtrial/model/Ride.java
deleted file mode 100644
index 1740eea..0000000
--- a/cross-ride2/src/com/crossover/techtrial/model/Ride.java
+++ /dev/null
@@ -1,158 +0,0 @@
-/**
- * 
- */
-package com.crossover.techtrial.model;
-
-import java.io.Serializable;
-import javax.persistence.Column;
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.GenerationType;
-import javax.persistence.Id;
-import javax.persistence.JoinColumn;
-import javax.persistence.ManyToOne;
-import javax.persistence.Table;
-import javax.validation.constraints.NotNull;
-
-@Entity
-@Table(name = "ride")
-public class Ride implements Serializable{
-
-  private static final long serialVersionUID = 9097639215351514001L;
-
-  @Id
-  @GeneratedValue(strategy = GenerationType.IDENTITY)
-  Long id;
-
-  @NotNull
-  @Column(name = "start_time")
-  String startTime;
-  
-  @NotNull
-  @Column(name = "end_time")
-  String endTime;
-  
-  @Column(name = "distance")
-  Long distance;
-  
-  @ManyToOne
-  @JoinColumn(name = "driver_id", referencedColumnName = "id")
-  Person driver;
-  
-  @ManyToOne
-  @JoinColumn(name = "rider_id", referencedColumnName = "id")
-  Person rider;
-
-  public Long getId() {
-    return id;
-  }
-
-  public void setId(Long id) {
-    this.id = id;
-  }
-
-  public String getStartTime() {
-    return startTime;
-  }
-
-  public void setStartTime(String startTime) {
-    this.startTime = startTime;
-  }
-
-  public String getEndTime() {
-    return endTime;
-  }
-
-  public void setEndTime(String endTime) {
-    this.endTime = endTime;
-  }
-
-  public Long getDistance() {
-    return distance;
-  }
-
-  public void setDistance(Long distance) {
-    this.distance = distance;
-  }
-
-  public Person getDriver() {
-    return driver;
-  }
-
-  public void setDriver(Person driver) {
-    this.driver = driver;
-  }
-
-  public Person getRider() {
-    return rider;
-  }
-
-  public void setRider(Person rider) {
-    this.rider = rider;
-  }
-  
-  
-
-  @Override
-  public int hashCode() {
-    final int prime = 31;
-    int result = 1;
-    result = prime * result + ((distance == null) ? 0 : distance.hashCode());
-    result = prime * result + ((driver == null) ? 0 : driver.hashCode());
-    result = prime * result + ((endTime == null) ? 0 : endTime.hashCode());
-    result = prime * result + ((id == null) ? 0 : id.hashCode());
-    result = prime * result + ((rider == null) ? 0 : rider.hashCode());
-    result = prime * result + ((startTime == null) ? 0 : startTime.hashCode());
-    return result;
-  }
-
-  @Override
-  public boolean equals(Object obj) {
-    if (this == obj)
-      return true;
-    if (obj == null)
-      return false;
-    if (getClass() != obj.getClass())
-      return false;
-    Ride other = (Ride) obj;
-    if (distance == null) {
-      if (other.distance != null)
-        return false;
-    } else if (!distance.equals(other.distance))
-      return false;
-    if (driver == null) {
-      if (other.driver != null)
-        return false;
-    } else if (!driver.equals(other.driver))
-      return false;
-    if (endTime == null) {
-      if (other.endTime != null)
-        return false;
-    } else if (!endTime.equals(other.endTime))
-      return false;
-    if (id == null) {
-      if (other.id != null)
-        return false;
-    } else if (!id.equals(other.id))
-      return false;
-    if (rider == null) {
-      if (other.rider != null)
-        return false;
-    } else if (!rider.equals(other.rider))
-      return false;
-    if (startTime == null) {
-      if (other.startTime != null)
-        return false;
-    } else if (!startTime.equals(other.startTime))
-      return false;
-    return true;
-  }
-
-  @Override
-  public String toString() {
-    return "Ride [id=" + id + ", startTime=" + startTime + ", endTime=" + endTime + ", distance=" + distance + ", driver=" + driver + ", rider=" + rider + "]";
-  }
-  
-  
-  
-}
diff --git a/cross-ride2/src/com/crossover/techtrial/repositories/PersonRepository.java b/cross-ride2/src/com/crossover/techtrial/repositories/PersonRepository.java
deleted file mode 100644
index a45a6eb..0000000
--- a/cross-ride2/src/com/crossover/techtrial/repositories/PersonRepository.java
+++ /dev/null
@@ -1,20 +0,0 @@
-/**
- * 
- */
-package com.crossover.techtrial.repositories;
-
-import java.util.List;
-import java.util.Optional;
-import org.springframework.data.repository.PagingAndSortingRepository;
-import org.springframework.data.rest.core.annotation.RestResource;
-import com.crossover.techtrial.model.Person;
-
-/**
- * Person repository for basic operations on Person entity.
- * @author crossover
- */
-@RestResource(exported=false)
-public interface PersonRepository extends PagingAndSortingRepository<Person, Long> {
-  Optional<Person> findById(Long id);
-  List<Person> findTop5();
-}
diff --git a/cross-ride2/src/com/crossover/techtrial/repositories/RideRepository.java b/cross-ride2/src/com/crossover/techtrial/repositories/RideRepository.java
deleted file mode 100644
index 6d61bf1..0000000
--- a/cross-ride2/src/com/crossover/techtrial/repositories/RideRepository.java
+++ /dev/null
@@ -1,20 +0,0 @@
-/**
- * 
- */
-package com.crossover.techtrial.repositories;
-
-import com.crossover.techtrial.model.Ride;
-
-import java.util.Optional;
-
-import org.springframework.data.repository.CrudRepository;
-import org.springframework.data.rest.core.annotation.RestResource;
-
-/**
- * @author crossover
- *
- */
-@RestResource(exported = false)
-public interface RideRepository extends CrudRepository<Ride, Long> {
-public Optional<Ride> findById(long rideId);
-}
diff --git a/cross-ride2/src/com/crossover/techtrial/service/PersonService.java b/cross-ride2/src/com/crossover/techtrial/service/PersonService.java
deleted file mode 100644
index 1ca59b9..0000000
--- a/cross-ride2/src/com/crossover/techtrial/service/PersonService.java
+++ /dev/null
@@ -1,23 +0,0 @@
-/**
- * 
- */
-package com.crossover.techtrial.service;
-
-import java.util.List;
-import com.crossover.techtrial.model.Person;
-
-/**
- * PersonService interface for Persons.
- * @author cossover
- *
- */
-public interface PersonService {
-  public List<Person> getAll();
-  
-  public List<Person> getTop5Person();
-  
-  public Person save(Person p);
-  
-  public Person findById(Long personId);
-  
-}
diff --git a/cross-ride2/src/com/crossover/techtrial/service/PersonServiceImpl.java b/cross-ride2/src/com/crossover/techtrial/service/PersonServiceImpl.java
deleted file mode 100644
index 4ea8720..0000000
--- a/cross-ride2/src/com/crossover/techtrial/service/PersonServiceImpl.java
+++ /dev/null
@@ -1,55 +0,0 @@
-/**
- * 
- */
-package com.crossover.techtrial.service;
-
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Optional;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Service;
-import com.crossover.techtrial.model.Person;
-import com.crossover.techtrial.repositories.PersonRepository;
-
-/**
- * @author crossover
- *
- */
-@Service
-public class PersonServiceImpl implements PersonService{
-
-  @Autowired
-  PersonRepository personRepository;
-  
-  /* (non-Javadoc)
-   * @see com.crossover.techtrial.service.PersonService#getAll()
-   */
-  @Override
-  public List<Person> getAll() {
-    List<Person> personList = new ArrayList<>();
-    personRepository.findAll().forEach(personList::add);
-    return personList;
-    
-  }
-  
-  public Person save(Person p) {
-    return personRepository.save(p);
-  }
-
-  @Override
-  public Person findById(Long personId) {
-    Optional<Person> dbPerson = personRepository.findById(personId);
-    return dbPerson.orElse(null);
-  }
-
-@Override
-public List<Person> getTop5Person() {
-	// TODO Auto-generated method stub
-	List<Person> personList = new ArrayList<>();
-    personRepository.findTop5().forEach(personList::add);
-    return personList;
-}
-  
-  
-
-}
diff --git a/cross-ride2/src/com/crossover/techtrial/service/RideService.java b/cross-ride2/src/com/crossover/techtrial/service/RideService.java
deleted file mode 100644
index 4926f6b..0000000
--- a/cross-ride2/src/com/crossover/techtrial/service/RideService.java
+++ /dev/null
@@ -1,20 +0,0 @@
-/**
- * 
- */
-package com.crossover.techtrial.service;
-
-import com.crossover.techtrial.model.Ride;
-
-/**
- * RideService for rides.
- * @author crossover
- *
- */
-public interface RideService {
-  
-  public Ride save(Ride ride);
-  
-  public Ride findById(Long rideId);
-  
-
-}
diff --git a/cross-ride2/src/com/crossover/techtrial/service/RideServiceImpl.java b/cross-ride2/src/com/crossover/techtrial/service/RideServiceImpl.java
deleted file mode 100644
index 4a647cc..0000000
--- a/cross-ride2/src/com/crossover/techtrial/service/RideServiceImpl.java
+++ /dev/null
@@ -1,33 +0,0 @@
-/**
- * 
- */
-package com.crossover.techtrial.service;
-
-import java.util.Optional;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Service;
-import com.crossover.techtrial.model.Ride;
-import com.crossover.techtrial.repositories.RideRepository;
-
-/**
- * @author crossover
- *
- */
-@Service
-public class RideServiceImpl implements RideService{
-
-  @Autowired
-  RideRepository rideRepository;
-  
-  public Ride save(Ride ride) {
-    return rideRepository.save(ride);
-  }
-  
-  public Ride findById(Long rideId) {
-    Optional<Ride> optionalRide = rideRepository.findById(rideId);
-    if (optionalRide.isPresent()) {
-      return optionalRide.get();
-    }else return null;
-  }
-
-}
diff --git a/cross-ride2/src/main/resources/db/changelog/db.changelog-1.0.xml b/cross-ride2/src/main/resources/db/changelog/db.changelog-1.0.xml
deleted file mode 100644
index 6073c06..0000000
--- a/cross-ride2/src/main/resources/db/changelog/db.changelog-1.0.xml
+++ /dev/null
@@ -1,37 +0,0 @@
-<?xml version="1.1" encoding="UTF-8" standalone="no"?>
-<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
-    <changeSet author="crossover" id="1001122001_create_tables">
-        <createTable tableName="person">
-            <column autoIncrement="true" name="id" type="BIGINT">
-                <constraints primaryKey="true" primaryKeyName="pk_person"/>
-            </column>
-            <column name="name" type="VARCHAR(255)"/>
-            <column name="email" type="VARCHAR(255)"/>
-            <column name="registration_number" type="VARCHAR(255)"/>
-        </createTable>
-        <createTable tableName="ride">
-            <column autoIncrement="true" name="id" type="BIGINT">
-                <constraints primaryKey="true" primaryKeyName="pk_ride"/>
-            </column>
-            <column name="driver_id" type="BIGINT"/>
-            <column name="rider_id" type="BIGINT"/>
-            <column name="start_time" type="VARCHAR(255)"/>
-            <column name="end_time" type="VARCHAR(255)"/>
-            <column name="distance" type="BIGINT"/>
-        </createTable>
-         <addForeignKeyConstraint baseColumnNames="driver_id"
-            baseTableName="ride"
-            constraintName="fk_address_person_driver"
-            onDelete="CASCADE"
-            onUpdate="RESTRICT"
-            referencedColumnNames="id"
-            referencedTableName="person"/>
-          <addForeignKeyConstraint baseColumnNames="rider_id"
-            baseTableName="ride"
-            constraintName="fk_address_person_rider"
-            onDelete="CASCADE"
-            onUpdate="RESTRICT"
-            referencedColumnNames="id"
-            referencedTableName="person"/>
-    </changeSet>
-</databaseChangeLog>
diff --git a/cross-ride2/src/main/resources/db/changelog/db.changelog-master.xml b/cross-ride2/src/main/resources/db/changelog/db.changelog-master.xml
deleted file mode 100644
index ea61ac5..0000000
--- a/cross-ride2/src/main/resources/db/changelog/db.changelog-master.xml
+++ /dev/null
@@ -1,6 +0,0 @@
-<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
-  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
-                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
-  <include file="db/changelog/db.changelog-1.0.xml"/>
-</databaseChangeLog>
\ No newline at end of file
diff --git a/cross-ride2/src/resources/application.properties b/cross-ride2/src/resources/application.properties
deleted file mode 100644
index bc7a1f6..0000000
--- a/cross-ride2/src/resources/application.properties
+++ /dev/null
@@ -1,15 +0,0 @@
-#PLEASE TRY NOT TO CHANGE SETTINGS AND USE APPROPRIATE ENV VARIABLES and DATABASE NAME AS CONFIGURED BELOW
-#Database connection Parameters
-spring.datasource.url: jdbc:mysql://localhost:3306/crossride?createDatabaseIfNotExist=true&useSSL=false
-spring.datasource.username=root
-spring.datasource.password=password
-spring.datasource.driver-class-name=com.mysql.jdbc.Driver
-
-spring.jpa.hibernate.ddl-auto=none
-
-spring.liquibase.change-log=classpath:/db/changelog/db.changelog-master.xml
-spring.data.rest.base-path=/api
-
-logging.level.root=INFO
-logging.level.org.hibernate.SQL=INFO
-logging.level.org.springframework=INFO
\ No newline at end of file
diff --git a/cross-ride2/src/resources/logback-spring.xml b/cross-ride2/src/resources/logback-spring.xml
deleted file mode 100644
index fb397ea..0000000
--- a/cross-ride2/src/resources/logback-spring.xml
+++ /dev/null
@@ -1,11 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<configuration>
-    <include resource="org/springframework/boot/logging/logback/defaults.xml" />
-    <property name="LOG_FILE" value="${LOG_FILE:-${LOG_PATH:-${LOG_TEMP:-${java.io.tmpdir:-/tmp}}/}spring.log}"/>
-    <include resource="org/springframework/boot/logging/logback/file-appender.xml" />
-    <include resource="org/springframework/boot/logging/logback/console-appender.xml" />
-    <root level="INFO">
-        <appender-ref ref="FILE" />
-        <appender-ref ref="CONSOLE"/>
-    </root>
-</configuration>
\ No newline at end of file
